### PR TITLE
Use latest pytorch-lightning version

### PR DESCRIPTION
**Pull request by hstonec**
Previously, we had fixed the version to an earlier release, which became incompatible after pip updated to 24.1. So, we took this opportunity to update it to the latest version.

Test plan:
All tests from github

### CODE REVIEW

File: dev-requirements.txt [hunk 1]
```diff
@@ -20,7 +20,8 @@ pyre-extensions
 pyre-check
 pytest
 pytest-cov
-pytorch-lightning==1.5.10
+pytorch-lightning
+tensorboard
```
Comment (yikaiMeta) [inline dev-requirements.txt:24]: why do we need this?

**Comment by Sanjay-Ganeshan** [comment]
Instead of using the latest version of pytorch lightning - can we set the specific version that happens to be the latest at the moment, with an `==` of `>=`?

That's 2.3.1, right now

https://pypi.org/project/pytorch-lightning/#history

Otherwise, LGTM

**Review by Sanjay-Ganeshan — APPROVED** [review]
Consider setting `pytorch-lightning==2.3.1`

Otherwise, LGTM

File: dev-requirements.txt (hunk 1 above)
Comment (hstonec) [reply dev-requirements.txt:24]: The latest pytorch-lightning needs that: 

image

**Comment by hstonec** [comment]
> Instead of using the latest version of pytorch lightning - can we set the specific version that happens to be the latest at the moment, with an `==` of `>=`?
> 
> That's 2.3.1, right now
> 
> https://pypi.org/project/pytorch-lightning/#history
> 
> Otherwise, LGTM

Yeah, we can do that if that can reduce some maintenance work in the future.

**Pull request merged**